libs/perun/services/src/lib/auth.service.ts

```diff
@@ -149,9 +148,22 @@ export class AuthService {
   private handleAuthCallback(): Promise<boolean> {
     return this.completeAuthentication().then(() => {
       const redirectUrl = sessionStorage.getItem('auth:redirect');
+      const params = sessionStorage.getItem('auth:queryParams').split('&');
+      let queryParams: Params = {};
+      if(params.length === 3 && (redirectUrl === '/' || redirectUrl === '/profile')){
+        queryParams = {
+          i: params[0].substr(2),
```

Comment:
It would be nice if this would work in general, if it would set correctly all queryParams, not only the ones specified here. You should be able just to split the values by the `=` character and set them all.